File: doc-examples/example-kotlin/src/main/kotlin/example/Place.kt
package example

import io.micronaut.serde.annotation.Serdeable
import io.micronaut.serde.annotation.Serdeable.Deserializable
import io.micronaut.serde.annotation.Serdeable.Serializable

@Serdeable
data class Place(
        @Deserializable(using = ReversePointSerde::class) // <1>
        @Serializable(using = ReversePointSerde::class) // <2>
        val point: Point,

        @Serdeable.Serializable(using = example.ReversePointSerde::class)
        val pointCustomSer: Point,

        @Deserializable(using = ReversePointSerde::class)
        val pointCustomDes: Point
)

File: doc-examples/example-python/src/main/python/example/Place.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.serde.annotation import Serdeable

from example.Point import Point
from example.ReversePointSerde import ReversePointSerde


@Serdeable
@dataclass
class Place:
    point: Annotated[
        Point,
        Serdeable.Serializable(using=ReversePointSerde),  # <1>
        Serdeable.Deserializable(using=ReversePointSerde),  # <2>
    ]
    pointCustomSer: Annotated[Point, Serdeable.Serializable(using=ReversePointSerde)]
    pointCustomDes: Annotated[Point, Serdeable.Deserializable(using=ReversePointSerde)]
